Fix out of bounds error from incrementing msg_i too soon

omarmciver opened the pull request:
Fix for index out of bounds error in prompt_templates/factory.py

Fixes: #10864

omarmciver committed:
Fix out of bounds error from incrementing msg_i too soon

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/BerriAI/litellm?pullRequest=10865) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/BerriAI/litellm?pullRequest=10865) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/BerriAI/litellm?pullRequest=10865) it.</sub>

krrishdholakia commented:
Hi @omarmciver how were you able to trigger this error?

omarmciver commented:
I'm actually not sure. I only make calls to litellm.acompletion and it came up in the stack trace. When I opened the code it just looked like a mistake that I've made many times myself.

I switch between openai, anthropic and ollama as my providers and I use tools with all after using the support call to test if tools are supported for a given model.

b-reyn commented:
@krrishdholakia  I'm able to reproduce this issue using `orieg/gemma3-tools:27b-it-qat` on ollama.

smrati commented:
@omarmciver thanks for suggesting the fix.
I have documented it on crewAI community forum here
https://community.crewai.com/t/anyone-managed-to-use-mcp-server-with-ollama-model-provider/6427/3?u=smrati_katiyar

omarmciver committed:
fix: Update test to expect keyword arguments in _regenerate_cli_key mock

The test was failing because the mock expected positional arguments but the actual implementation uses keyword arguments. Updated the test to properly handle keyword arguments.

omarmciver committed:
fix: apply formatter to fix linting issues

- Remove trailing whitespace
- Ensure proper formatting with Black
- Fix indentation issues
